packages/core/src/payment/payment-request-transformer.ts

```diff
@@ -144,4 +146,21 @@ export default class PaymentRequestTransformer {
                 hostedFormNonce: nonce,
             };
     }
+
+    private _mapShippingAddress(checkoutState: InternalCheckoutSelectors, paymentMethod?: PaymentMethod): Address | undefined {
+        if (
```

Comment:
I think we discussed to use `initializationData` here for making the decision to resolve address or not. And we can also pass a property to decide if address is required or not.

Reply:
I know, but we cannot determine that on the back-end because we rely on what the user selects, picking up the entire order, shipping, or a combination of them (in the future).